conda_build/metadata.py

```diff
@@ -605,6 +543,72 @@ def binary_has_prefix_files(self):
     def skip(self):
         return self.get_value('build/skip', False)
 
+    def _get_contents(self, permit_undefined_jinja):
+        '''
+        Get the contents of our [meta.yaml|conda.yaml] file.
+        If jinja is installed, then the template.render function is called
+        before standard conda macro processors.
+    
+        permit_undefined_jinja: If True, *any* use of undefined jinja variables will
+                                evaluate to an emtpy string, without emitting an error.
+        '''
+        try:
+            import jinja2
+        except ImportError:
+            print("There was an error importing jinja2.", file=sys.stderr)
+            print("Please run `conda install jinja2` to enable jinja template support", file=sys.stderr)
+            with open(self.meta_path) as fd:
+                return fd.read()
+    
+        from conda_build.jinja_context import context_processor
+    
+        path, filename = os.path.split(self.meta_path)
+        loaders = [# search relative to '<conda_root>/Lib/site-packages/conda_build/templates'
+                   jinja2.PackageLoader('conda_build'),
+                   # search relative to RECIPE_DIR
+                   jinja2.FileSystemLoader(path)
+                   ]
+    
+        # search relative to current conda environment directory
+        conda_env_path = os.environ.get('CONDA_DEFAULT_ENV')  # path to current conda environment
+        if conda_env_path and os.path.isdir(conda_env_path):
+            conda_env_path = os.path.abspath(conda_env_path)
+            conda_env_path = conda_env_path.replace('\\', '/') # need unix-style path
+            env_loader = jinja2.FileSystemLoader(conda_env_path)
+            loaders.append(jinja2.PrefixLoader({'$CONDA_DEFAULT_ENV': env_loader}))
+    
+        undefined_type = jinja2.StrictUndefined
+        if permit_undefined_jinja:
```

Comment:
>I see you didn't actually write this bit

Right, this isn't part of the current PR -- it was just relocated from elsewhere in the file.  I did not write the lines above (related to jinja loaders).  But as luck would have it, I did write the `UndefinedNeverFail` class below (in an earlier PR).

>for instance, what would happen if I removed the `__int__` strangeness...

Without the `__int__ = lambda _: 0` line, templates like `{{ int(PKG_VERSION) }}`, would raise an error during the first pass (since `PKG_VERSION` isn't defined until the second pass).  (Side note: I see `__int__` is accidentally set twice in this code... no harm there, but I'll fix it now.)